lib/ecto_ldap/adapter.ex

```diff
@@ -469,161 +466,203 @@ defmodule Ecto.Ldap.Adapter do
     :eldap.mod_replace(convert_to_erlang(attribute), [value])
   end
 
-
   @doc """
-  Convert a value from its data-store representation to something that Ecto expects.
-  `:id` fields are always returned unchanged.
 
-  ### Examples
+  Retrieves a function that can convert a value from the LDAP
+  data format to a form that Ecto can interpret.
 
-      iex> Ecto.Ldap.Adapter.load(:id, "uid=jeff.weiss,ou=users,dc=example,dc=com")
-      {:ok, "uid=jeff.weiss,ou=users,dc=example,dc=com"}
+  The functions (or types) returned by these calls will be invoked
+  by Ecto while performing the translation of values for LDAP records.
 
-      iex> Ecto.Ldap.Adapter.load(:id, 123456)
-      {:ok, 123456}
 
-      iex> Ecto.Ldap.Adapter.load(:id, nil)
-      {:ok, nil}
+  `:id` fields and unrecognized types are always returned unchanged.
+
+  ### Examples
 
+    iex> Ecto.Ldap.Adapter.loaders(:id, :id)
+    [:id]
+
+    iex> Ecto.Ldap.Adapter.loaders(:woo, :woo)
+    [:woo]
 
   Given that LDAP uses ASN.1 GeneralizedTime for its datetime storage format, values
   where the type is `Ecto.DateTime` will be converted to a string and parsed as ASN.1
   GeneralizedTime, assuming UTC ( `"2016040112[34[56[.789]]]Z"` )
 
   ### Examples
 
-      iex> Ecto.Ldap.Adapter.load(:datetime, ['20160401123456.789Z'])
-      {:ok, {{2016, 4, 1}, {12, 34, 56, 789000}}}
-
-      iex> Ecto.Ldap.Adapter.load(:datetime, ['20160401123456Z'])
-      {:ok, {{2016, 4, 1}, {12, 34, 56, 0}}}
-
-      iex> Ecto.Ldap.Adapter.load(:datetime, ['201604011234Z'])
-      {:ok, {{2016, 4, 1}, {12, 34, 0, 0}}}
-
-      iex> Ecto.Ldap.Adapter.load(:datetime, ['2016040112Z'])
-      {:ok, {{2016, 4, 1}, {12, 0, 0, 0}}}
+    iex(1)> conversion_function = hd(Ecto.Ldap.Adapter.loaders(:datetime, :datetime))
```

Comment:
I explicitly don't want `hd` here. I think it decreases the readability and conceals what's returned.
I'd rather it be `[conversion_function] = ...`

Also, kill the counters in `iex>`